### PR TITLE
Only allow MFA'ed users to be invited in access groups.

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

File: mozillians/groups/forms.py [hunk 1]
```diff
@@ -127,9 +128,21 @@ def clean(self):
...
         is_manager = user.userprofile.is_manager
         is_curator = self.instance.curators.filter(id=user.userprofile.id).exists()
+
+        error_msgs = []
         if not is_curator and not is_manager:
             msg = _(u'You need to be the curator of this group before inviting someone to join.')
-            self._errors['invites'] = self.error_class([msg])
+            error_msgs.append(msg)
+
+        if self.instance.is_access_group and not self.instance.name == settings.NDA_GROUP:
```
Comment (akatsoulas) [inline mozillians/groups/forms.py]: Shouldn't this check be true only for ajax requests?

File: mozillians/users/views.py [hunk 2]
```diff
@@ -100,6 +105,32 @@ def get_autocomplete_location_query(qs, q):
     return qs.filter(name__icontains=q)
 
 
+class AccessGroupInvitationAutocomplete(autocomplete.Select2QuerySetView):
```
Comment (akatsoulas) [inline mozillians/users/views.py]: Let's break this `Select2QuerySetView` into two. One for nda members and one for employees. The NDA one can inherit from the employees. This way we can use the employees `Select2QuerySetView` for the curators invite form.

File: mozillians/groups/forms.py (hunk 1 above)
Comment (johngian) [reply mozillians/groups/forms.py]: No this happens on form submission.

**Comment by johngian** [comment]
@akatsoulas PR updated

File: mozillians/groups/forms.py [hunk 3]
```diff
@@ -127,9 +128,21 @@ def clean(self):
...
+            msg = _(u'Only staff and NDA members are allowed to be invited')
+            for profile in self.cleaned_data['invites']:
+                if not profile.is_nda or not profile.can_create_access_groups:
+                    error_msgs.append(msg)
+                    break
+
+        if error_msgs:
+            self._errors['invites'] = self.error_class(error_msgs)
             del self.cleaned_data['invites']
```
Comment (akatsoulas) [inline mozillians/groups/forms.py]: Is this needed here?

File: mozillians/users/models.py [hunk 4]
```diff
@@ -427,6 +427,20 @@ def is_public(self):
     def is_manager(self):
         return self.user.is_superuser or self.user.groups.filter(name='Managers').exists()
 
+    @property
+    def is_mfa(self):
```
Comment (akatsoulas) [inline mozillians/users/models.py]: Since this is not used anywhere, let's drop it.

File: mozillians/users/views.py [hunk 5]
```diff
@@ -100,6 +105,39 @@ def get_autocomplete_location_query(qs, q):
...
+        if not self.request.user.userprofile.is_vouched:
+            return UserProfile.objects.none()
+
+        queries = []
+
+        # Query staff profiles
+        for domain in settings.AUTO_VOUCH_DOMAINS:
+            pks = IdpProfile.objects.filter(
+                email__endswith='@' + domain).values_list('profile__pk', flat=True)
+            queries.append(Q(pk__in=pks))
```
Comment (akatsoulas) [inline mozillians/users/views.py]: super nit - feel free to ignore. Maybe a bit more cryptic/pythonic (depends on personal taste).

```python
idp_query = reduce(operator.or_, [Q(email__endswith='@' + doamain) for  domain in settings.AUTO_VOUCH_DOMAINS])
idp_pks = IdpProfile.objects.filter(**idp_query)
```

**Pull request merged**